src/App.tsx

```diff
@@ -1,34 +1,86 @@
 /* eslint-disable max-len */
-import React from 'react';
+
+// default imports
+
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+// components
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+// outer function
+
+import { getTodos } from './api';
+import { filterTodos } from './utils/filterTodos';
+
+// types
+
+import { Todo } from './types/Todo';
+import { FilterByState } from './types/FilterByState';
+
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo>(null);
+  const [filterBy, setFilterBy] = useState<FilterByState>(FilterByState.ALL);
+  const [query, setQuery] = useState('');
+  const [hasErrorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodoList)
+      .catch(e => setErrorMessage(e))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filteredTodos = filterTodos(todoList, filterBy, query);
+
   return (
     <>
       <div className="section">
         <div className="container">
+          {hasErrorMessage && (
+            <h1 className="title">Error has occured while downloading toods</h1>
```

Comment:
There is a typo in the error message: 'toods' should be 'todos'. Ensure that error messages are clear and free of typos to improve user experience.